src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -349,30 +350,45 @@ void LoopManager::fuse_loop_ports(std::vector<LoopPort>& output_ports,
 }
 
 void LoopManager::update_loop_ports(const ExpressionPtr& expr) {
-    auto output_ports = expr->get_output_ports();
-    for (size_t i = 0; i < expr->get_input_count(); ++i) {
-        const auto& source = expr->get_input_port_connector(i)->get_source();
-        const auto common_outer_loop_ids = get_common_outer_loops(expr, source.get_expr());
-        // The source output port can have several consumers (including the current expr) that can be potential output ports
-        // So we should verify on the possible future output ports
-        size_t count_of_common_outer_loops = common_outer_loop_ids.size();
-        for (const auto& source_consumer : source.get_connected_ports()) {
-            if (source_consumer.get_expr() == expr)
+    auto update_ports = [&](const ov::snippets::lowered::ExpressionPort& connected_port) {
+        const auto is_output = connected_port.get_type() == ExpressionPort::Output;
+        // Iterate through all Loops of the connected expression
+        for (const auto& loop_id : connected_port.get_expr()->get_loop_ids()) {
+            const auto& loop_info = get_loop_info(loop_id);
+            // If the connected expression port is not Loop port - nothing to update
+            // If the target expression is not from the same Loop - nothing to update
+            if (!loop_info->is_loop_port(connected_port) || !is_loop_id_found(expr, loop_id))
                 continue;
-            count_of_common_outer_loops = std::min(count_of_common_outer_loops, get_common_outer_loops(source.get_expr(), source_consumer.get_expr()).size());
-        }
-        replace_loop_ports({common_outer_loop_ids.cbegin(), common_outer_loop_ids.cbegin() + count_of_common_outer_loops}, source, output_ports);
-        // Save previous port
-        if (count_of_common_outer_loops != common_outer_loop_ids.size()) {
-            output_ports.insert(output_ports.begin(), source);
-            replace_loop_ports({common_outer_loop_ids.cbegin() + count_of_common_outer_loops, common_outer_loop_ids.cend()}, source, output_ports);
+
+            std::vector<ExpressionPort> new_ports;
+            // Check if some ports of target expression must be Loop port
+            const auto target_expr_ports = is_output ? expr->get_output_ports() : expr->get_input_ports();
+            for (const auto& port : target_expr_ports) {
+                if (utils::should_be_loop_port(port, loop_id))
+                    new_ports.push_back(port);
+            }
+            // Leave the connected expression port as Loop port if needed
+            if (utils::should_be_loop_port(connected_port, loop_id))
+                new_ports.push_back(connected_port);
+
+            // Nothing should be updated
+            if (new_ports.size() == 1 && new_ports.front() == connected_port)
+                continue;
+
+            loop_info->replace_with_new_ports(connected_port, new_ports);
```

Comment:
I understand your concern about complexity (but it just looks so). However, this safe method fully covers all our needs here :blush:
This method is implemented to easily update affected loop ports when we insert expression to LinearIR (currently, we use it only in `insert_expr` helpers). It allows you to just call `loop_manager->update_loop_ports(inserted_expr)` to update need loops.
We iterate through input and outputs of the expression, take loops of consumers/sources and check if the current port - is Loop port. And replace existing port with new one if needed.

As for, `O((num_inputs + num_outputs) * num_loop_ids * (num_loop_ports + num_loop_ids))`, the values of this variables are 1-3. I mean that this value `O(n)` is not so big.

In my opinion, this method cannot be improved because you should exactly know which existing LoopPorts should be updated (you need to iterate through consumers and sources and iterate through their loop_ids) and with which new ports they should be replaced (input and output ports of the target expression).